ci/docker/Dockerfile.alpine-release

```diff
@@ -0,0 +1,51 @@
+FROM alpine:3.15 as build
+
+# creates /opensmtpd dir and makes all following commands to run in it
+# https://docs.docker.com/engine/reference/builder/#workdir
+WORKDIR /opensmtpd
+
+# install necessary packages
+RUN apk add --no-cache \
+    autoconf \
+    automake \
+    bison \
+    ca-certificates \
+    fts-dev \
+    gcc \
+    fts \
+    libevent-dev \
+    libtool \
+    libtool \
+    linux-pam-dev \
+    make \
+    musl-dev \
+    openssl \
+    openssl-dev \
+    zlib-dev
```

Comment:
Last thing, if we start providing an "official" Dockerfile I'd prefer if it didn't contain the packages needed to build.  (have to admit that the Dockerfile I'm currently using keeps the build-dependency because I'm lazy ;-)

How hard would it be to change this into a multi-stage build?  We'd need to copy `smtpd`, `smtpctl`, `smtp`, the libexecs inside `/usr/local/libexec/opensmtpd` and re-create some directories under `/var` in the clean image, then reinstall libevent, openssl/libressl.

Reply:
I phrased my initial comment poorly, sorry.  I wasn't asking you to actually do this, rather hear your opinion in how much difficult / fragile it would be.

You guessed right, with a [multi-stage build](https://docs.docker.com/build/building/multi-stage/) I meant to use a first temporary image to build OpenSMTPD (along with all the build-depends needed) and then copy the binaries into a separate image.  Recent-ish docker version make this somewhat easy, although for an application like OpenSMTPD it could be tricky as it's not a single binary.

It's just a "what if".

Reply:
Thanks for introducing me to this, I will try to implement this shortly.

Reply:
Implemented multi-stage build.